base/libgit2/tag.jl

```diff
@@ -35,6 +35,7 @@ function name(tag::GitTag)
 end
 
 function target(tag::GitTag)
-    oid_ptr = Ref(ccall((:git_tag_target_id, :libgit2), Ptr{Oid}, (Ptr{Void}, ), tag.ptr))
-    return oid_ptr[]
+    oid_ptr = ccall((:git_tag_target_id, :libgit2), Ptr{Oid}, (Ptr{Void}, ), tag.ptr)
+    oid_ptr == C_NULL && return Oid()
```

Comment:
no, problem would occur long before this call, so throwing exception will not give anything, only complicate handing of this function (unnecessary try-catch). However, empty Oid is a harmless thing whenever you use it.